File: src/main/java/org/example/menu/commands/editing/Remove.java
package org.example.menu.commands.editing;

import org.example.data.FileManager;
import org.example.menu.Command;
import org.example.products.SouvenirCollection;
import org.example.utility.Config;

import java.util.Scanner;

public class Remove implements Command {
    @Override
    public void execute() {
        System.out.println("*** Удалить производителя и сувениры. ***");
        SouvenirCollection souvenirCollection = new SouvenirCollection();
        souvenirCollection.deleteManufacturerAndSouvenirs(getManufacturerName());
        saveChange(souvenirCollection);
    }

    private String getManufacturerName() {
        System.out.println("Введите название производителя: ");
        Scanner scanner = new Scanner(System.in);
        return scanner.nextLine();
    }

    private void saveChange(SouvenirCollection souvenirCollection) {
        FileManager.saveManufacturersToFile(
                Config.DEFAULT_MANUFACTURER_TEXT_FILE,
                souvenirCollection.getManufacturerList());
        FileManager.saveSouvenirsToFile(
                Config.DEFAULT_SOUVENIR_TEXT_FILE,
                souvenirCollection.getSouvenirList()
        );
        System.out.println("Изменения сохранены.");
    }
}

File: src/main/java/org/example/menu/commands/editing/Remove.kt
package org.example.menu.commands.editing

import org.example.data.FileManager
import org.example.menu.Command
import org.example.products.SouvenirCollection
import org.example.utility.Config

class Remove : Command {
    override fun execute() {
        println("*** Удалить производителя и сувениры. ***")
        val souvenirCollection = SouvenirCollection()
        souvenirCollection.deleteManufacturerAndSouvenirs(getManufacturerName())
        saveChange(souvenirCollection)
    }

    private fun getManufacturerName(): String {
        println("Введите название производителя: ")
        return readlnOrNull() ?: ""
    }

    private fun saveChange(souvenirCollection: SouvenirCollection) {
        FileManager.saveManufacturersToFile(
            Config.DEFAULT_MANUFACTURER_TEXT_FILE,
            souvenirCollection.manufacturerList
        )
        FileManager.saveSouvenirsToFile(
            Config.DEFAULT_SOUVENIR_TEXT_FILE,
            souvenirCollection.souvenirList
        )
        println("Изменения сохранены.")
    }
}